Dev

galmalka122 opened the pull request.

galmalka122 committed:
Add initial implementation of PowerPoint presentation analysis and OpenAI
integration

galmalka122 committed:
added basic api implementation for upload files and get their status

galmalka122 committed:
added explainer script - runs indefinitely and scans the uploads directory to find unclarified pptx files
implemented as a module

galmalka122 committed:
added explainer script - runs indefinitely and scans the uploads directory to find unclarified pptx files
implemented as a module

galmalka122 committed:
added __init__ to act as separate module
added comments
status route modified

galmalka122 committed:
Merge remote-tracking branch 'final-exercise-galmalka122/api' into dev

galmalka122 committed:
Merge pull request #1 from Scaleup-Excellenteam/api

Api

galmalka122 committed:
Merge remote-tracking branch 'final-exercise-galmalka122/dev' into dev

galmalka122 committed:
final product v0.1

excellenteam-scaleup commented on pptx_clarifier/__init__.py:
```diff
@@ -0,0 +1,34 @@
+import asyncio
+import os
+import sys
+import threading
```
why do you work with threads?
the heavy long part is the GPT part, which is dealt using asyncio

galmalka122 replied:
i couldn't figure out how to run both clarifier and the api

excellenteam-scaleup commented on pptx_clarifier/client/Status.py:
```diff
@@ -0,0 +1,48 @@
+import datetime
+
+
+class Status:
```
Good

excellenteam-scaleup commented on pptx_clarifier/definitions.py:
```diff
@@ -0,0 +1,10 @@
+root = sys.path[1]
+uploads_directory = os.path.join(root, "pptx_clarifier", "uploads")
+outputs_directory = os.path.join(root, "pptx_clarifier", "outputs")
+os.makedirs(uploads_directory, exist_ok=True)
```
Nice flag

excellenteam-scaleup commented on pptx_clarifier/definitions.py:
```diff
@@ -0,0 +1,10 @@
+os.makedirs(uploads_directory, exist_ok=True)
+os.makedirs(outputs_directory, exist_ok=True)
+
+base_url = 'localhost:5000'
```
didn't I already see that somewhere?

excellenteam-scaleup commented on pptx_clarifier/logger.py:
Nice

excellenteam-scaleup commented on pptx_clarifier/pptx_clarifier_api/web_api.py:
```diff
@@ -0,0 +1,114 @@
+
+
+@app.get('/upload')
+def upload_file():
```
what is this for?

galmalka122 replied:
it is for testing purposes which i forgot to delete :/ , and discovered posting file using postman right after :|

excellenteam-scaleup commented on pptx_clarifier/pptx_explainer/__init__.py:
Not sure the init file is where you want to save your prompt

excellenteam-scaleup commented on pptx_clarifier/pptx_explainer/__init__.py:
```diff
@@ -0,0 +1,36 @@
+###
+
+
+Instructions: 1. Please wait for the first slide before generating explanations. The slide data will be provided 
```
Interesting approach.
Does this prompt synchronized with the demand to run it a-synchronically? we don't want GPT to have dependencies between slides which will make it wait and take longer..
I might give GPT more context and less instructions.

galmalka122 replied:
hi :)
though the gpt responses are a-sync the requests being sent synchronize (each slide is sent to gpt and while the program waits to the response, it sends another slides, but in order), so the prompt being sent for each slide includes the previous slides in it.

excellenteam-scaleup commented on pptx_clarifier/pptx_explainer/explainer.py:
```diff
@@ -0,0 +1,80 @@
+    logger.info(f"Processing {file_name}")
+    try:
+        json_path = await clarify(file_name)
+    except FileNotFoundError as file_not_found_error:
```
Nice error handling

excellenteam-scaleup commented on pptx_clarifier/pptx_explainer/presentation_parser.py:
```diff
@@ -0,0 +1,120 @@
+        raise ValueError(f'{path} is not a pptx file')
+
+
+if __name__ == '__main__':
```
have a main function please

excellenteam-scaleup reviewed (approved):
Good job.
Seems like you put a lot of work into this system.
The code looks good, and well separated into modules and functions.
I think you sometimes have unneeded documentation, and should think whether the documentation really adds information that is not covered by the function's name.
Please also respond to the comment about your prompt, and good luck in the next stage!

galmalka122 committed:
final commit.
added db using sqlite and sqlalchemy
users can now add email for more convenience tracking

galmalka122 committed:
fixed pull request notes

excellenteam-scaleup reviewed (approved):
Better to have new PR for each exercise, but good fixes to the comments.
